src/basic/ansi-color.c

```diff
@@ -83,6 +85,45 @@ static ColorMode get_color_mode_impl(void) {
         return COLOR_256;
 }
 
+bool validate_ansi_color(const char *color) {
+        int r;
+
+        if (isempty(color))
+                return true;
+
+        if (string_has_cc(color, /* ok= */ "\x1B"))
+                return false;
```

Comment:
the color field should just be the "inner" part of the ansi sequence, i.e. not allow \x1b in it. instead we prefix this ourselves on output.